### PR TITLE
DB-11141: return a warning if no rows are affected by a DML

### PR DESCRIPTION
**Pull request by jyuanca**

### CODE REVIEW

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2292/

**Review by msirek — APPROVED** [review]

**Review by jpanko1 — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

File: splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/SQLWarningIT.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        new TableCreator(conn)
+                .withCreate("create table A (c1 int)")
+                .withInsert("insert into A values(?)")
+                .withRows(rows(
+                        row(1),
+                        row(2))).create();
+    }
+
+    @Test
+    public void TestNoRowsAffectedWarning() throws Exception {
```
Comment (dgomezferro) [inline splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/SQLWarningIT.java:47]: can you add a negative test case with `splice.db2.error.compatible` set to false?

Comment (dgomezferro) [reply splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/SQLWarningIT.java:47]: method name should start with lowercase

**Review by dgomezferro — CHANGES_REQUESTED** [review]

**Comment by ascend1** [comment]
@jyuanca Can you please take a look at @dgomezferro 's comment?

**Review by carolp-503 — APPROVED** [review]
Additional changes look good to me.

**Pull request merged**